# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/PresShell.cpp
# Commit: a4ef62ad447f
# Full Hash: a4ef62ad447f3bdeca250ebc7052854226c8891f
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-05-09 04:15:00
# Regressor Bug: 1932150
# File Overlap Count: 1
# Description:
#   Bug 1932150 - Allow using flat tree order for point comparing in selection r=jjaschke,smaug,dom-core
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D231588
# ==============================================================================

diff -r fb611166c7ab -r a4ef62ad447f layout/base/PresShell.cpp
--- a/layout/base/PresShell.cpp	Thu May 08 13:37:20 2025 +0000
+++ b/layout/base/PresShell.cpp	Thu May 08 13:37:20 2025 +0000
@@ -5063,7 +5063,8 @@
         // if the node is within the range, append it to the temporary list
         bool before, after;
         nsresult rv =
-            RangeUtils::CompareNodeToRange(content, aRange, &before, &after);
+            RangeUtils::CompareNodeToRange<TreeKind::ShadowIncludingDOM>(
+                content, aRange, &before, &after);
         if (NS_SUCCEEDED(rv) && !before && !after) {
           itemToInsert = i;
           bool snap;
